tauri-updater/src/lib.rs

```diff
@@ -1,19 +1,1034 @@
+    extract_path = extract_path
+      .parent()
+      .map(PathBuf::from)
+      .expect("Unable to find the extract path")
+      .parent()
+      .map(PathBuf::from)
+      .expect("Unable to find the extract path")
+  };
+
+  // We should use APPIMAGE exposed env variable
+  // This is where our APPIMAGE should sit and should be replaced
+  if target == "linux" && env::var_os("APPIMAGE").is_some() {
+    extract_path = PathBuf::from(env::var_os("APPIMAGE").expect("Unable to extract APPIMAGE path"))
+  }
+
+  extract_path
+}
+
+// Return the archive type to save on disk
+fn detect_archive_in_url(path: &str, target: &str) -> String {
+  path
+    .split('/')
+    .next_back()
+    .unwrap_or(&archive_name_by_os(target))
+    .to_string()
+}
+
+// Fallback archive name by os
+// The main objective is to provide the right extension based on the target
+// if we cant extract the archive type in the url we'll fallback to this value
+fn archive_name_by_os(target: &str) -> String {
+  let archive_name = match target {
+    "darwin" | "linux" => "update.tar.gz",
+    _ => "update.zip",
+  };
+  archive_name.to_string()
+}
+
+// Convert base64 to string and prevent failing
+fn base64_to_string(base64_string: &str) -> crate::Result<String> {
+  let decoded_string = &decode(base64_string.to_owned())?;
+  let result = from_utf8(&decoded_string)?.to_string();
+  Ok(result)
+}
+
+// Validate signature
+// need to be public because its been used
+// by our tests in the bundler
+pub fn verify_signature(
+  archive_path: &PathBuf,
+  release_signature: String,
+  pub_key: &str,
+) -> crate::Result<bool> {
+  // we need to convert the pub key
+  let pub_key_decoded = &base64_to_string(pub_key)?;
+  let public_key = PublicKey::decode(pub_key_decoded);
+  if public_key.is_err() {
+    bail!(
+      crate::Error::Updater,
+      "Something went wrong with pubkey decode"
+    )
+  }
+
+  let public_key_ready = public_key.expect("Something wrong with the public key");
+
+  let signature_decoded = base64_to_string(&release_signature)?;
+
+  let signature =
+    Signature::decode(&signature_decoded).expect("Something wrong with the signature");
+
+  // We need to open the file and extract the datas to make sure its not corrupted
+  let file_open = OpenOptions::new()
+    .read(true)
+    .open(&archive_path)
+    .expect("Can't open our archive to validate signature");
+
+  let mut file_buff: BufReader<File> = BufReader::new(file_open);
+
+  // read all bytes since EOF in the buffer
+  let mut data = vec![];
+  file_buff
+    .read_to_end(&mut data)
+    .expect("Can't read buffer to validate signature");
+
+  // Validate signature or bail out
+  public_key_ready.verify(&data, &signature)?;
+  Ok(true)
+}
+
+#[cfg(test)]
+mod test {
+  use super::*;
+  #[cfg(target_os = "macos")]
+  use std::env::current_exe;
+  #[cfg(target_os = "macos")]
+  use std::fs::File;
+  #[cfg(target_os = "macos")]
+  use std::path::Path;
+
+  macro_rules! aw {
```

Comment:
maybe make this a little more descriptive than just `aw!`.  Something like `block!`